lib/perl/Genome/VariantReporting/Suite/BamReadcount/MaxVafObservedInterpreter.pm

```diff
@@ -49,14 +59,13 @@ sub _interpret_entry {
 
     my %normal_vafs;
     my %tumor_vafs;
-    my @sample_name_accessors = qw/normal_sample_names tumor_sample_names/;
-    my @vaf_hash_names        = (\%normal_vafs,        \%tumor_vafs);
-    my $it                    = each_array(@sample_name_accessors, @vaf_hash_names);
-    while ( my ($sample_name_accessor, $vaf_hash_ref) = $it->() ) {
+    my @sample_name_accessors     = qw/normal_sample_names    tumor_sample_names/;
+    my @vaf_hash_names            = (\%normal_vafs,           \%tumor_vafs);
+    my @vaf_interpreter_accessors = qw/normal_vaf_interpreter tumor_vaf_interpreter/;
+    my $it                    = each_array(@sample_name_accessors, @vaf_hash_names, @vaf_interpreter_accessors);
+    while ( my ($sample_name_accessor, $vaf_hash_ref, $vaf_interpreter_accessor) = $it->() ) {
         if ($self->$sample_name_accessor) {
-            my $interpreter = Genome::VariantReporting::Suite::BamReadcount::VafInterpreter->create(
-                sample_names => [$self->$sample_name_accessor],
-            );
+            my $interpreter = $self->$vaf_interpreter_accessor;
```

Comment:
I'm trying to avoid this by checking for `$self->$sample_name_accessor` on line 67. Do you think that's sufficient or should I add another check for the return value of `$self->$vaf_interpreter_accessor`?